validator/client/attest.go

```diff
@@ -254,9 +254,11 @@ func (v *validator) waitOneThirdOrValidBlock(ctx context.Context, slot types.Slo
 	defer span.End()
 
 	// Don't need to wait if requested slot is the same as highest valid slot.
+	v.highestValidSlotLock.Lock()
 	if slot <= v.highestValidSlot {
 		return
```

Comment:
Returning here can result in deadlock since the mutex was not released

Reply:
Apologies and good catch. Just pushed a fix.

Reply:
All good! Thanks so much!